pagopa/io-app
# fix: [IOPID-1546] Fix services refresh token error

## Short description
This PR fixes the missing services refresh token handling.

It takes the chance to refactor the `withRefreshApiCall` in terms of its input parameters so that it can be called like that:
- `withRefreshApiCall(apiCall(...apiCallParams), action)`
- `withRefreshApiCall(apiCall(...apiCallParams), { errorMessage: "foo"})`
- `withRefreshApiCall(apiCall(...apiCallParams), { skipErrorMessage: true})`
- `withRefreshApiCall(apiCall(...apiCallParams))`

<details open><summary>Demo</summary>
<p>

| ✅ | ❌ |
| - | - |
| <video src="https://github.com/pagopa/io-app/assets/16268789/1a46653f-0e83-487d-ba42-f3684a72277e" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/fe4929dd-1434-457c-a7a4-2971756d9e56" /> |


</p>
</details> 

## List of changes proposed in this pull request
* [ts/features/fastLogin/saga/utils/index.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Ffeatures%2FfastLogin%2Fsaga%2Futils%2Findex.ts)
* [ts/sagas/profile.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fprofile.ts)
* [ts/sagas/services/servicePreference/handleGetServicePreferenceSaga.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fservices%2FservicePreference%2FhandleGetServicePreferenceSaga.ts)
* [ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fservices%2FservicePreference%2FhandleUpsertServicePreferenceSaga.ts)
* [ts/sagas/startup/__tests__/loadServiceDetailRequestHandler.test.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fstartup%2F__tests__%2FloadServiceDetailRequestHandler.test.ts)
* [ts/sagas/startup/__tests__/loadVisibleServicesHandler.test.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fstartup%2F__tests__%2FloadVisibleServicesHandler.test.ts)
* [ts/sagas/startup/loadServiceDetailRequestHandler.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fstartup%2FloadServiceDetailRequestHandler.ts)
* [ts/sagas/startup/loadVisibleServicesHandler.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fstartup%2FloadVisibleServicesHandler.ts)
* [ts/sagas/user/userMetadata.ts](https://github.com/pagopa/io-app/blob/7193af46b2117472731391dbaa73827e48593bd3/ts%2Fsagas%2Fuser%2FuserMetadata.ts)

## How to test
Run the app against the dev-server with Fast Login enabled. Go to services and wait for the session to expire. Then navigate to a service detail screen. The token refresh saga should be triggered.
